### PR TITLE
Fix meter/instrument filtering in MetricsEventSource

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Review by noahfalk — APPROVED** [review]
👍

File: src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs [hunk 1]
```diff
@@ -649,12 +649,6 @@ private void ParseSpecs(string? metricsSpecs)
                     return;
                 }
 
-                if (metricsSpecs.Length == 0)
-                {
-                    _aggregationManager!.IncludeAll();
-                    return;
-                }
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs:656]: This was an invalid special-casing?

Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs:656]: That's correct. In .NET 9.0, we added support for filtering emitted metrics data. Initially, we considered adding this special case for compatibility with .NET 8.0, especially after a test failed without it. However, we found that this caused all instruments to be listened to, which wasn't the exact behavior in 8.0. Removing this case made us fully compatible with 8.0, and we fixed the test that was failing for a different reason.

**Comment by sebastienros** [comment]
I confirm this is fixing the regressions with HttpClient instances. I am not seeing "msquic" events anymore after it either.

**Comment by tarekgh** [comment]
Thanks @sebastienros for confirming!

**Pull request merged**